### PR TITLE
[Claim] Reject tx modals

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Merges into `claim-reject-tx`

Example 1:
image

Example 2:
image

# To Test

1. try claim direct (free claim)
2. reject in wallet, see modal

approve
1. try approve
2. reject in wallet
3. see modal

### CODE REVIEW

**Comment by W3stside** [comment]
that transparency is just from the screenshot

File: src/custom/hooks/useErrorMessageAndModal.tsx [hunk 1]
```diff
@@ -23,3 +23,30 @@ export default function useErrorMessage() {
...
+    error && openModal()
+  }
+
+  return {
+    error: internalError,
+    handleCloseError,
+    handleSetError,
+    ErrorModal: ({ message = internalError }: { message?: string }) => (
+      <TransactionErrorModal onDismiss={handleCloseError} message={message} />
+    ),
```
Comment (alfetopito) [inline src/custom/hooks/useErrorMessageAndModal.tsx:50]: should we have a `useCallback` for this component too like `TransactionErrorModal`?

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx [hunk 2]
```diff
@@ -57,7 +57,7 @@ export default function InvestOption({ approveData, updateInvestAmount, claim }:
 
   const onMaxClick = useCallback(() => {
     // TODO: remove
```
Comment (alfetopito) [inline src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:59]: Remove?

**Review by alfetopito — APPROVED** [review]
👍

File: src/custom/hooks/useErrorMessageAndModal.tsx (hunk 1 above)
Comment (W3stside) [reply src/custom/hooks/useErrorMessageAndModal.tsx:50]: ah, yes good catch

File: src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx (hunk 2 above)
Comment (W3stside) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:59]: NO

Comment (W3stside) [reply src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx:59]: yes

File: src/custom/hooks/useErrorMessageAndModal.tsx [hunk 3]
```diff
@@ -23,3 +23,30 @@ export default function useErrorMessage() {
       error ? <SwapCallbackError showClose={showClose} handleClose={handleCloseError} error={error} {...rest} /> : null,
   }
 }
+
+export function useErrorModal() {
+  // Any async bc errors
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/custom/hooks/useErrorMessageAndModal.tsx:32]: removed, was blockchain

**Review by anxolin — APPROVED** [review]

**Pull request merged**